tests/product/base_product_case.py

```diff
@@ -92,12 +92,20 @@ class BaseProductTestCase(BaseTestCase):
                                   r'|\nWarning: (\[%(host)s] )?Timed out ' \
                                   r'trying to connect to %(host)s \(tried 1 ' \
                                   r'time\)\n\nUnderlying exception:' \
-                                  r'\n    timed out\n)'
+                                  r'\n    timed out\n)' \
```

Comment:
Can you add a comment at the top of these regexes to explain what they are, since they're a bit hard to read if you don't already know.